core/src/main/java/org/infinispan/remoting/transport/impl/SingleTargetRequest.java

```diff
@@ -27,7 +26,7 @@ public class SingleTargetRequest<T> extends AbstractRequest<T> {
 
    // Only changes from non-null to null
    @GuardedBy("responseCollector")
```

Comment:
This is no longer true

Reply:
Partly: one branch actually does go through this. I removed the comment anyway